org.osgi.test.assertj.framework/src/test/java/org/osgi/test/assertj/servicereference/AbstractServiceReferenceAssertTest.java

```diff
@@ -0,0 +1,87 @@
+package org.osgi.test.assertj.servicereference;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.assertj.core.api.AssertFactory;
+import org.junit.jupiter.api.Test;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.ServiceReference;
+import org.osgi.framework.Version;
+import org.osgi.test.assertj.testutil.AbstractAssertTest;
+
+public abstract class AbstractServiceReferenceAssertTest<SELF extends AbstractServiceReferenceAssert<SELF, ServiceReference<?>>>
+	extends AbstractAssertTest<SELF, ServiceReference<?>> {
+	protected AbstractServiceReferenceAssertTest(AssertFactory<ServiceReference<?>, SELF> assertThat) {
+		super(assertThat);
+	}
+
+	@Test
+	public void isAssignableTo() throws Exception {
+
+		ServiceReference<?> sr = mock(ServiceReference.class);
+		setActual(sr);
+
+		when(sr.isAssignableTo(null, null)).thenReturn(true);
+		assertPassing("isNot", x -> aut.isAssignableTo(null, null), null);
+
+		when(sr.isAssignableTo(null, null)).thenReturn(false);
+		assertFailing("is", x -> aut.isAssignableTo(null, null), null)
+			.hasMessageMatching("(?si).*to be assignable to:.*but it was not.*");
```

Comment:
Better to pass in more sensible values here and then check that the message contains those values. At the moment we might be putting the wrong information into the error message (or omitting it) and we wouldn't know. Additional advantage is that you don't have to change the mock method's return value - just pass in different values.

Reply:
done